src/harness/testcases/WINNF_FT_S_GPR_testcase.py

```diff
@@ -0,0 +1,458 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+import os
+import sas
+import sas_testcase
+from sas_test_harness import SasTestHarnessServer, generateCbsdRecords, \
+    generatePpaRecords
+from util import addCbsdIdsToRequests, winnforum_testcase, configurable_testcase, writeConfig, \
+  loadConfig, getRandomLatLongInPolygon, makePpaAndPalRecordsConsistent
+
+
+class GwpzProtectionTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_GPR_1_default_config(self, filename):
+    """ Generates the WinnForum configuration for GPR.1. """
+
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    # Load devices info
+    device_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    # Moving device_1 to a location within 40 KMs of GWPZ zone
+    device_1['installationParam']['latitude'] = 37.6161
+    device_1['installationParam']['longitude'] = -106.95261
+
+    device_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    # Moving device_2 to a location outside 40 KMs of GWPZ zone
+    device_2['installationParam']['latitude'] = 37.83075
+    device_2['installationParam']['longitude'] = -107.65025
+
+    device_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    # Moving device_3 to a location inside GWPZ zone
+    device_3['installationParam']['latitude'], \
+    device_3['installationParam']['longitude'] = getRandomLatLongInPolygon(gwpz_record_1)
+
+    device_4 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_d.json')))
```

Comment:
Since we already have device_2 outside 40kms, isn't it enough? Why do we need 2 devices outside 40km?

Comment also applies for device_4 in GPR.2